evm/src/fuzz/invariant/executor.rs

```diff
@@ -479,6 +478,36 @@ impl<'a> InvariantExecutor<'a> {
     }
 }
 
+/// Collects data from call for fuzzing. However, it first verifies that the sender is not an EOA
+/// before inserting it into the dictionary. Otherwise, we flood the dictionary with
+/// randomly generated addresses.
+fn collect_data(
+    state_changeset: &mut HashMap<Address, revm::Account>,
+    sender: &Address,
+    call_result: &RawCallResult,
+    fuzz_state: EvmFuzzState,
+) {
+    // Verify it has no code.
+    let mut has_code = false;
+    if let Some(Some(code)) = state_changeset.get(sender).map(|account| account.info.code.as_ref())
+    {
+        has_code = !code.is_empty();
+    }
+
+    // We keep the nonce changes to apply later.
+    let mut sender_changeset = None;
+    if !has_code {
+        sender_changeset = state_changeset.remove(sender);
+    }
```

Comment:
do we want to do this? what if there's e.g. a smart contract wallet that would make a call and hit e.g. a `if isContract` check that would trigger a re-entrancy via onERC721/onERC1155 fallback?

Reply:
This code only prevents adding this address to the dictionary through the `statechangeset`. The changes are still applied. So, unless I misunderstood something, it wouldn't prevent that scenario.